lua_cjson.c

```diff
@@ -74,6 +74,8 @@
 #define DEFAULT_DECODE_INVALID_NUMBERS 0
 #endif
 
+#define MT_AS_ARRAY "MT_AS_ARRAY"
```

Comment:
Because this string is used as a key in the global Lua registry, we need a more specific name like `cjson_mt_arr`.  What do you think?